datamodel_code_generator/parser/jsonschema.py

```diff
@@ -633,10 +632,21 @@ def parse_raw_obj(self, name: str, raw: Dict[str, Any], path: List[str]) -> None
         self.parse_ref(obj, path)
 
     def parse_raw(self) -> None:
-        raw_obj: Dict[Any, Any] = json.loads(self.text)  # type: ignore
+        raw_obj: Dict[Any, Any] = yaml.safe_load(self.text)  # type: ignore
         obj_name = raw_obj.get('title', 'Model')
         obj_name = self.model_resolver.add([], obj_name, unique=False).name
         self.parse_raw_obj(obj_name, raw_obj, [])
-        definitions = raw_obj.get('definitions', {})
+        definitions, path = get_oas_schema(raw_obj)
         for key, model in definitions.items():
-            self.parse_raw_obj(key, model, ['#/definitions', key])
+            self.parse_raw_obj(key, model, [path, key])
+
+
+def get_oas_schema(oas_spec: Dict[Any, Any]):
```

Comment:
Support for different OAS schemas:
OAS2 - #/definitions
OAS3- #/components/schemas
...

Reply:
Sorry, I forgot the difference.
This file is for `JSON Schema`.
this method name should not have `OSA`.
Would you please remove `OSA` from the method ?

Reply:
I just noticed you had another class for OAS3, and I moved out stuff there.

We'll find a way to fix the usage of `definitions` in jsonschema another time :)

Reply:
This code generator supported only OSA3 for the first time.
But, I supported JSON Schema by a request.

We will fix it next time:wink: